discovery_engine_core/core/src/engine.rs

```diff
@@ -1092,6 +1090,115 @@ mod tests {
         );
     }
 
+    #[tokio::test]
+    async fn test_update_stack_no_error_when_no_stack_is_ready() {
+        let mut mock_ops = new_mock_stack_ops();
+        mock_ops
+            .expect_new_items()
+            .returning(|_, _, _| Err(NewItemsError::NotReady));
+
+        let stack_ops = vec![Box::new(mock_ops) as BoxedOps];
+        let mut stacks = create_stacks_from_stack_ops(stack_ops);
+
+        let market = ("US", "en");
+        let mut no_op_ranker = new_no_op_ranker();
+        let mut exploration_stack = exploration::Stack::new(StackData::default()).unwrap();
+
+        let result = update_stacks(
+            &mut stacks,
+            &mut exploration_stack,
+            &mut no_op_ranker,
+            &[],
+            10,
+            10,
+            10,
+            &market.into(),
+        )
+        .await;
+
+        assert!(result.is_ok());
+    }
+
+    #[tokio::test]
+    async fn test_update_stack_no_error_when_one_stack_is_successful() {
+        let mut mock_ops_ok = new_mock_stack_ops();
+        mock_ops_ok
+            .expect_new_items()
+            .returning(|_, _, _| Ok(vec![new_mock_article()]));
+
+        let mut mock_ops_failed = new_mock_stack_ops();
+        mock_ops_failed
+            .expect_new_items()
+            .returning(|_, _, _| Err(NewItemsError::Error("mock_ops_failed_error".into())));
+
+        let stack_ops = vec![
+            Box::new(mock_ops_ok) as BoxedOps,
+            Box::new(mock_ops_failed) as BoxedOps,
+        ];
+        let mut stacks = create_stacks_from_stack_ops(stack_ops);
+
+        let market = ("US", "en");
+        let mut no_op_ranker = new_no_op_ranker();
+        let mut exploration_stack = exploration::Stack::new(StackData::default()).unwrap();
+
+        let result = update_stacks(
+            &mut stacks,
+            &mut exploration_stack,
+            &mut no_op_ranker,
+            &[],
+            10,
+            10,
+            10,
+            &market.into(),
+        )
+        .await;
+
+        result.unwrap();
```

Comment:
I do prefer `result.unwrap()` because it actually displays the error message in case of an error, instead of just a vague "expectation failed". Are there any major downsides to this that I'm not seeing?